### PR TITLE
Improve neo4j docker compose

### LINKED ISSUE #3962: task: Improve docker compose around database service
### Component

CI/CD

### Task Description

Docker compose could be improved for the NEO4J authentication to take into account database password ...

### PR DESCRIPTION
**Pull request by BaptisteGi**
- Craft a proper `NEO4J_AUTH` env variable using the `INFRAHUB_DB_PASSWORD` variable

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -129,9 +129,9 @@ services:
     image: ${NEO4J_DOCKER_IMAGE:-neo4j:5.19.0-community}
     restart: unless-stopped
     environment:
-      - "NEO4J_AUTH=neo4j/admin"
-      - "NEO4J_dbms_security_procedures_unrestricted=apoc.*"
-      - "NEO4J_dbms_security_auth__minimum__password__length=4"
+      NEO4J_AUTH: neo4j/${INFRAHUB_DB_PASSWORD}
```
Comment (fatih-acar) [inline docker-compose.yml]:
```suggestion
      NEO4J_AUTH: ${INFRAHUB_DB_USERNAME:-neo4j}/${INFRAHUB_DB_PASSWORD:-admin}
```

You should use the same syntax as above in the x-infrahub-config block, to keep default values when the env var is not set.

Comment (BaptisteGi) [reply docker-compose.yml]: For some reason we can't change NEO4J login ... https://neo4j.com/docs/operations-manual/current/docker/introduction/#docker-auth
(I'll set default password, shouldn't it be 'infrahub' tho?)

File: docker-compose.yml [hunk 2]
```diff
@@ -129,9 +129,9 @@ services:
...
+      dbms_security_procedures_unrestricted: apoc.*"
+      dbms_security_auth__minimum__password__length: 4
```
Comment (BeArchiTek) [inline docker-compose.yml]: https://neo4j.com/docs/operations-manual/current/docker/ref-settings/
you need to keep `NEO4J_` in those 2

File: docker-compose.yml (hunk 1 above)
Comment (BaptisteGi) [reply docker-compose.yml]: I adjusted the part with password, for the login I don't know .... So far it sounds pretty much like a trap as if the user changes the var `INFRAHUB_DB_USERNAME` it just won't work ... Should we add a comment like "If you use neo4j please don't change that" or make sure that whrn using neo4j this var doesn't exist ...

Comment (fatih-acar) [reply docker-compose.yml]: Right I didn't know about that thanks for the pointer. Then I think we can leave it as-is...

File: docker-compose.yml [hunk 3]
```diff
@@ -129,9 +129,9 @@ services:
     image: ${NEO4J_DOCKER_IMAGE:-neo4j:5.19.0-community}
     restart: unless-stopped
     environment:
-      - "NEO4J_AUTH=neo4j/admin"
-      - "NEO4J_dbms_security_procedures_unrestricted=apoc.*"
-      - "NEO4J_dbms_security_auth__minimum__password__length=4"
+      NEO4J_AUTH: neo4j/${INFRAHUB_DB_PASSWORD:-admin}
+      dbms_security_procedures_unrestricted: apoc.*"
```
Comment (fatih-acar) [inline docker-compose.yml]: Why did you remove the NEO4J_ prefixes?

**Review by BeArchiTek — CHANGES_REQUESTED** [review]

**Comment by BaptisteGi** [comment]
BTW do I need to apply the same change to the other places (development folder)?

**Review by BeArchiTek — APPROVED** [review]
